app/modules/ads/module.js

```diff
@@ -101,62 +103,76 @@ class Ads {
   }
 
   configureBillTheLizard(instantGlobals) {
-    const { context, slotService } = window.Wikia.adEngine;
-    const { billTheLizard } = window.Wikia.adServices;
+    const { context } = window.Wikia.adEngine;
+
+    if (context.get('bidders.prebid.bidsRefreshing.enabled')) {
+      this.cheshirecatCalled = false;
+      this.cheshirecatConfig = instantGlobals.wgAdDriverBillTheLizardConfig || {};
+      this.cheshirecatPredictions = {};
+
+      context.set('bidders.prebid.bidsRefreshing.bidsBackHandler', this.callCheshireCat.bind(this));
+      context.push('listeners.slot', {
+        onRenderEnded: (adSlot) => {
+          if (adSlot.config.slotName === 'incontent_boxad_1' && this.biddersCalled && !this.cheshirecatCalled) {
+            this.callCheshireCat();
+          }
+        },
+      });
+    }
+  }
 
+  callCheshireCat() {
     function getNextIncontentId(predictions) {
       return `incontent_boxad_${Object.keys(predictions).length + 2}`;
     }
 
-    if (context.get('bidders.prebid.bidsRefreshing.enabled')) {
-      this.cheshirecatPredictions = {};
+    const { context, slotService } = window.Wikia.adEngine;
+    const { billTheLizard } = window.Wikia.adServices;
 
-      context.set('bidders.prebid.bidsRefreshing.bidsBackHandler', () => {
-        const config = instantGlobals.wgAdDriverBillTheLizardConfig || {};
-        const bidderPrices = targeting.getBiddersPrices('mobile_in_content');
-
-        context.set('services.billTheLizard.projects', config.projects);
-        context.set('services.billTheLizard.timeout', config.timeout || 0);
-        context.set('services.billTheLizard.parameters.cheshirecat', {
-          bids: [
-            bidderPrices.bidder_1 || 0,
-            bidderPrices.bidder_2 || 0,
-            0,
-            bidderPrices.bidder_4 || 0,
-            0,
-            bidderPrices.bidder_6 || 0,
-            bidderPrices.bidder_7 || 0,
-            0,
-            bidderPrices.bidder_9 || 0,
-            bidderPrices.bidder_10 || 0,
-            bidderPrices.bidder_11 || 0,
-            bidderPrices.bidder_12 || 0,
-            bidderPrices.bidder_13 || 0,
-            bidderPrices.bidder_14 || 0,
-            bidderPrices.bidder_15 || 0,
-            bidderPrices.bidder_16 || 0,
-          ].join(';'),
-        });
+    const bidderPrices = targeting.getBiddersPrices('mobile_in_content');
+
+    context.set('services.billTheLizard.projects', this.cheshirecatConfig.projects);
+    context.set('services.billTheLizard.timeout', this.cheshirecatConfig.timeout || 0);
+    context.set('services.billTheLizard.parameters.cheshirecat', {
```

Comment:
If we decide to separate module -- what do you think about separate method for that? i.e. `serializeBids(slotName)`

Reply:
Yep, done.